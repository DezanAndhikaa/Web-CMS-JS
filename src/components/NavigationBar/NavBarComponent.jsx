

import React from 'react';
import { AppBar, Toolbar } from '@material-ui/core';
import { UTLogoNew, AccountPic, CmsLogo } from 'assets/imgs';
import { MenuToggle } from 'assets/icons';
import isAccessTokenValid from 'core/HelpersFunction';
import './NavBarComponent.scss';
import PopUpMenu from 'components/PopUpMenu/PopUpMenu';
import { Menu, StorageKey } from '../../constants';
import LogoutModal from 'views/Logout/Logout';

class NavBarComponent extends React.Component {

	constructor(props) {
		super(props);
		this.state = {
		  isShowModal: false,
		  isShowModalLogOut: false
		};
	}

	isClicked = () => {
		this.setState({isShowModal: !this.state.isShowModal})
	}

	isClose = () => {
        this.setState({isShowModal: !this.state.isShowModal})
	}

	handleClick = (menu, tab) => {
		this.props.push({
		  pathname: menu,
		  whichTab: tab
		});
	}

	isShowModalLogout = () => {
		this.setState({
			isShowModal: !this.state.isShowModal,
			isShowModalLogOut: !this.state.isShowModalLogOut
		})
	}

	isClosed=() => {
		this.setState({
			isShowModalLogOut: !this.state.isShowModalLogOut,
			isShowModal: false
		})
	}

	handleLogout = () => {
		localStorage.removeItem(StorageKey.USER_DATA);
		this.props.logout();
		this.props.onLogout();
		this.isClosed();
		this.props.push(Menu.LOGIN);
	};
	
	renderPopUpMenu() {
		return(
			<PopUpMenu 
				{...this.props}				
				{...this.state}
				openModal={this.state.isShowModal}
				closemodal={this.isClose}
				onClickMenuLogout={this.isShowModalLogout}
			/>
		)
	}

	showModalLogout(){
		return(
			<LogoutModal 
                {...this.props}
                openModal={this.state.isShowModalLogOut}
                onYesClicked={this.handleLogout}
                onNoClicked={this.isClosed}
			/>
		)
	}

	render() {
		let navBar = null;
		if (isAccessTokenValid()) {
			if (this.props.displayMode === 'web') {
				navBar = (
					<AppBar position="fixed" className="app-bar">
						<Toolbar variant="dense" className="toolbar">
							<img src={UTLogoNew} alt="" className="logo-ut"/>
							<div className="info-login">
								<p>
									{`Hi, ${this.props.userData.firstName} ${this.props.userData.lastName}`}
								</p>
								<div onClick={this.isClicked}>
									<img src={AccountPic} className="account-pic" alt="" />
									{this.renderPopUpMenu()}
								</div>
								{this.state.isShowModalLogOut && this.showModalLogout()}
							</div>
						</Toolbar>
					</AppBar>
				);
			} else {
				navBar = (
					<AppBar position="fixed" className="app-bar-tab">
						<div onClick={() => this.props.toggleMenu(!this.props.menuDrawerState)}>
							<img src={MenuToggle} alt="menu" className="menu-toggle" />
						</div>
						<Toolbar variant="dense" className="toolbar">
							<img src={CmsLogo} className="logo-dca" alt="" />
						</Toolbar>
					</AppBar>
				);
			}
		}
		return navBar;
	}
}

export default NavBarComponent;
