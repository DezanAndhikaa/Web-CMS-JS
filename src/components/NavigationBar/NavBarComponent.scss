@import '../../themes/variables';
@import '../../themes/functions';

.badge{
    font-family: 'Lato-Bold';
    font-size: vw(14);
    border: 2px solid white;
}

.app-bar{
    display: flex;
    color: #000000 !important;
    background-color: #FFFFFF !important;
    box-shadow: none !important;
    height: vw(88) !important;
    flex-direction: column;
    position: relative;
    z-index: 0;
    .toolbar{
        padding: 0 !important;
        display: flex;
        flex-direction: row;
        .logo-ut{
            display: flex;
            height: vw(88) !important;
            margin-left: 0px;
            margin-right: vw(50);
        }
        // .right-side{
        //     display: flex;
        //     flex: 1;
        //     justify-content: flex-end;
        //     align-items: center;
            .info-login{
                display: flex;
                flex: 1;
                justify-content: flex-end;
                align-items: center;
                padding-right: vw(30);
                p{
                    font-family: 'Lato-Bold';
                    font-size: vw(20);
                    line-height: vw(24);
                    color: #3C3B3B;
                    margin-right: vw(20);
                }
                .account-pic{
                    height: vw(48) !important;
                }
            }
        // }
    }
}

.app-bar-tab{
    color: #000000 !important;
    background-color: #FFFFFF !important;
    box-shadow: none !important;
    display: flex;
    align-items: flex-end;
    justify-content: center;
    padding-right: vw-tab(30);
    height: vw-tab(88) !important;
    box-shadow: 0px vw-tab(4) vw-tab(10) rgba(0, 0, 0, 0.04) !important;
    .logo-dca{
        width: vw-tab(164);
        height: vw-tab(41);
    }
    .menu-toggle{
        position: absolute;
        top: 0;
        left: 0;
        width: vw-tab(33);
        height: vw-tab(33);
        top: vw-tab(28);
        left: vw-tab(30);
    }
}

@media (max-width: 599px) {
    .app-bar-tab{
        color: #000000 !important;
        background-color: #FFFFFF !important;
        box-shadow: none !important;
        display: flex;
        align-items: flex-end;
        justify-content: center;
        padding-right: vw-mobile(20);
        height: vw-mobile(73) !important;
        box-shadow: 0px vw-mobile(4) vw-mobile(10) rgba(0, 0, 0, 0.04) !important;
        .logo-dca{
            width: vw-mobile(119);
            height: vw-mobile(29);
        }
        .menu-toggle{
            position: absolute;
            top: 0;
            left: 0;
            width: vw-mobile(23);
            height: vw-mobile(23);
            top: vw-mobile(25);
            left: vw-mobile(20);
        }
    }
}