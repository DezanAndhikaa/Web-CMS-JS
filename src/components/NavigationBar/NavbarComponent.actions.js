export const ToggleMenuActionType = 'TOGGLE_MENU';

export function toggleMenuAction(isOpen) {
	return {
		type: ToggleMenuActionType,
		isOpen,
	};
}
