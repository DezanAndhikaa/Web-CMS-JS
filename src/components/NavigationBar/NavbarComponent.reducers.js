import { ToggleMenuActionType } from './NavbarComponent.actions';

const initialMenuIsOpen = false;

export default function toggleMenuReducer(state = initialMenuIsOpen, action) {
	if (action.type === ToggleMenuActionType) {
		return action.isOpen;
	}
	return state;
}
