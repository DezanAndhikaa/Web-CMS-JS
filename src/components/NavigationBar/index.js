import { connect } from 'react-redux';
import NavBarComponent from './NavBarComponent';
import { toggleMenuAction } from './NavbarComponent.actions';
import { push } from 'connected-react-router';
import { removeDataAction } from 'core/StorageHelper';
import { USER_DATA } from '../../constants';
import { logoutAction } from 'actions/planning';

const mapStateToProps = (state) => ({
	path: state.router.location.pathname,
	userData: state.userData,
	displayMode: state.displayMode,
	menuDrawerState: state.menuDrawerState,
});

const mapDispatchToProps = (dispatch) => ({
	toggleMenu: (isOpen) => dispatch(toggleMenuAction(isOpen)),
	push: (path) => dispatch(push(path)),
	logout: () => dispatch(removeDataAction(USER_DATA)),
	onLogout: () => dispatch(logoutAction()),
});

export default connect(mapStateToProps, mapDispatchToProps)(NavBarComponent);
