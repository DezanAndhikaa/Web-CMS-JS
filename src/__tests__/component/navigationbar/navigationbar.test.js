import React from 'react';
import Enzyme, { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import NavigationBar from '../../../components/NavigationBar/NavBarComponent';

Enzyme.configure({adapter:new Adapter()});
it("render error navigation bar component correctly", () => {
    const wrapper = shallow (
        <NavigationBar />
    )
    expect(wrapper).toMatchSnapshot();
})