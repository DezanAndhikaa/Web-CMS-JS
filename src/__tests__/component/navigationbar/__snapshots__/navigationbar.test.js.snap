// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render error navigation bar component correctly 1`] = `ShallowWrapper {}`;
